pkg/evaluators/metadata/generic_http.go

```diff
@@ -142,7 +142,11 @@ func (h *GenericHttp) buildRequest(ctx gocontext.Context, endpoint, authJSON str
 		if err != nil {
 			return nil, err
 		}
-		req.Header.Set(header.Name, fmt.Sprintf("%s", headerValue))
+		headerValueStr, err := json.StringifyJSON(headerValue)
```

Comment:
Is that _always_ true tho? Couldn't this be "just a plain string" ever?

Reply:
One thing we can do to mitigate the impact of this PR over the other affected sources for the conversion is to make `json.StringifyJSON` return earlier if the `interface{}` input is a `string`.

Reply:
Just pushed a commit: https://github.com/Kuadrant/authorino/pull/501/commits/113717d144f6d3349be47759bf6a63b3bc782e0d

Reply:
Ok, so it was _not_ always the case? And now a "plain string" remains a plain string... do I get this right? In anycase... I think looking into something like what's proposed in #500 wrt to `Value.ResolveFor` signature would be a good refactor for ... future us. Today's `interface{}` (and I know that's sourced in golang's type system) is just too broad I think. Further more the asymmetry of the function wrt its input & output is definitively a code smell to me.

Reply:
It was the case, I played with these functions a bit... I get what's going on now